tests/unit_tests/common/test_ldg_ptr.cu

```diff
@@ -0,0 +1,82 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2023, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <gridtools/common/ldg_ptr.hpp>
+
+#include <gtest/gtest.h>
+
+#include <cuda_test_helper.hpp>
+
+namespace gridtools {
+    namespace {
```

Comment:
Possibly, we could add some compile time checks (maybe to test `simple_ptr_holder`), but I guess they are implicitly tested via their usage in SIDs.

from `is_sid`:
```
                // `is_trivially_copyable` check is applied to the types that are will be passed from host to device
                std::is_trivially_copy_constructible<PtrHolder>,

                // verify that `PtrDiff` is sane
                std::is_default_constructible<PtrDiff>,
                std::is_convertible<decltype(std::declval<Ptr const &>() + std::declval<PtrDiff const &>()), Ptr>,

                // `PtrHolder` supports `+` as well
                // TODO(anstaf): we can do better here: verify that if we transform PtrHolder that way the result
                //               thing also models SID
                std::is_same<
                    std::decay_t<decltype(std::declval<PtrHolder const &>() + std::declval<PtrDiff const &>())>,
                    PtrHolder>,
```

Reply:
Yes, I think `simple_ptr_holder` is used in quite a few places, so it should indeed be tested quite thoroughly.